### PR TITLE
Add custom reason

### PR DESCRIPTION
**Pull request by sac-cas**
Added reason to responses so any status code can be mocked with a given reason

### CODE REVIEW

File: test_responses.py [hunk 1]
```diff
@@ -54,6 +56,34 @@ def run():
...
+    @responses.activate
+    def run():
+        responses.add(
+            responses.GET,
+            "http://example.com",
+            body=b"take care!",
+            status=299,
+            reason="warning",
+        )
+        requests.get("http://example.com")
```
Comment (hramezani) [inline test_responses.py:69]: I think we need to just test the newly added feature. something like:

```python
        resp = requests.get("http://example.com")
        assert resp.status_code == 299
        assert resp.reason == "warning"

```

File: test_responses.py [hunk 2]
```diff
@@ -54,6 +56,34 @@ def run():
...
+        assert len(responses.calls) == 1
+        assert responses.calls[0].request.url == "http://example.com/"
+        assert responses.calls[0].response.content == b"take care!"
+        assert responses.calls[0].response.reason == "warning"
+        assert responses.calls[0].response.status_code == 299
+
+        requests.get("http://example.com?foo=bar")
```
Comment (hramezani) [inline test_responses.py:76]: What is the purpose of this test?

File: test_responses.py [hunk 3]
```diff
@@ -127,6 +157,46 @@ def run():
...
+                method=responses.GET,
+                url="http://example.com/api",
+                body="replacement",
+                status=299,
+                reason="warning",
+            ),
+        ),
+    ],
+)
+def test_replace_with_reason(original, replacement):
```
Comment (hramezani) [inline test_responses.py:180]: I think we don't need this test.

**Comment by hramezani** [comment]
I don't know for sure if we want this or not. Other than that, we need to add [documentation to README.rst](https://github.com/getsentry/responses#response-parameters).

File: test_responses.py (hunk 2 above)
Comment (sacastells) [reply test_responses.py:76]: The test is similar to `test_response` just with custom status and reason. We can only test the newly added features. In my code I checked all values just to make an entire test.

File: test_responses.py (hunk 3 above)
Comment (sacastells) [reply test_responses.py:180]: Replace must be tested separately I think. The given `test_replace` adds more responses than necessary so I'm not sure what is its purpose.

File: test_responses.py (hunk 2 above)
Comment (sacastells) [reply test_responses.py:76]: The test is similar to `test_response` just with custom status and reason. We can only test the newly added features. In my code I checked all values just to make an entire test.

File: test_responses.py (hunk 1 above)
Comment (sacastells) [reply test_responses.py:69]: We can only test the newly added features. In my code I checked all values just to make an entire test. I think it's at least not worse to test the other values :)

File: test_responses.py (hunk 3 above)
Comment (sacastells) [reply test_responses.py:180]: Replace must be tested separately I think. The given test_replace adds more responses than necessary so I'm not sure what is its purpose.

**Comment by sacastells** [comment]
I only added the reason text to add/replace not to callback btw.

**Comment by markstory** [comment]
Is your usecase for this to enable status text on non-standard response codes? Or something else?

**Comment by sacastells** [comment]
> Is your usecase for this to enable status text on non-standard response codes? Or something else?

Yes, my only usecase is to enable custom status text for standard response codes. It can also be used for standard usecases but I think the majority will use the standard response text.

**Comment by carloshorn** [comment]
The proposed feature is already included in *responses*. You can define a custom reason using a `response_callback`.

Example:
```python
def custom_reason(response):
    if response.status_code == 299:
        response.reason = 'My Reason'
    return response

with responses.RequestsMock(response_callback=custom_reason) as reqmock:
    r = responses.Response(responses.GET, 'https://www.example.com', status=299)
    reqmock.add(r)
    print(requests.get('https://www.example.com').reason)
```

**Pull request closed without merging**